app/models/case_cycle.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# The cycle for subsidy cases
+class CaseCycle < UuidApplicationRecord
+  monetize :copay_cents
+
+  enum copay_frequency: { daily: 'daily', weekly: 'weekly', monthly: 'monthly' }
```

Comment:
As an alternative, what if we used a regular string field, and validated that the input is included in a constant array of values?

i.e.
```
class CaseCycle

FREQUENCIES = [ "daily", "weekly", "monthly" ]

validates :copay_frequency, inclusion: { in: FREQUENCIES }
```
This gives us the added benefit of using the defined constant outside of the model, so we can use the same list for specs, etc.

Thoughts? @weedySeaDragon @arku ?

Reply:
That's exactly what an `enum` is for:  so you have a structured, reusable way of saying "here is an enumerated list of things to use; you can only choose one of them".
WRT validations, the _specification_ should not re-use the enums, because then you're not really _specifying_ anything.  (it's kind of circular reasoning instead of explicitly saying "here are the items that can be in that enumerated list.)
But when you're testing the behavior (or internal stuff), it's definitely ok to use the enum itself.

If you need to reuse an `enum` across classes like we do with _license types,_  then we can do this (which I just happened to write a bit ago as I'm working thru the Subsidy Rules):
```
# frozen_string_literal: true

# License types
# These are the list of valid license types.
# These correspond to an ENUM type defined in the Postgres DB
class Lookups::LicenseType
  enum valid_types: { licensed_center: 'licensed_center',
                      licensed_family_home: 'licensed_family_home',
                      licensed_group_home: 'licensed_group_home',
                      license_exempt_home: 'license_exempt_home',
                      license_exempt_center: 'license_exempt_center' }
end
```

Here's an example of how it can be used in the Business class:
```
class Business
   ...
   enum license_types: Lookups::LicenseType.valid_types
  ...

```

Reply:
I think we covered this in Slack, but just to follow up here:  What you're writing as 'tests' are actually also _specifications._   Yes, you may be writing unit tests, but they also serve as a description/specification of what should happen.  (The "Spec" in "RSpec" means _specification_.  I can tell you the whole history sometime.  Spoiler: "xSpec" originates from a Smalltalk person: Kent Beck).

Someone is going to later read through the specs/tests to help them understand what is going on and how it is all supposed to be working.  (Code is written for people; bits are for computers.)

I do also agree that it's always a balance with design:  you need to make sure that future (and current!) developers will be able to maintain the code in a reasonable way.  With FOSS, the knowledge varies so _widely._